W3ChampionsIdentificationService/ServiceVerification/B2BVerificationService.cs

```diff
@@ -0,0 +1,42 @@
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+
+namespace W3ChampionsIdentificationService.ServiceVerification;
+
+public class B2BVerificationService
+{
+    private readonly SymmetricSecurityKey _key;
+    private readonly string _issuer;
+    private readonly string _audience;
+
+    public B2BVerificationService(string secret, string issuer)
+    {
+        _issuer = issuer;
+        _key = new SymmetricSecurityKey(System.Text.Encoding.ASCII.GetBytes(secret));
+        _audience = "w3c-identification-service";
+    }
+
+    public bool ValidateToken(string token)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var validationParameters = new TokenValidationParameters
+        {
+            ValidateIssuer = true,
+            ValidateAudience = true,
+            ValidateLifetime = true,
+            ValidateIssuerSigningKey = true,
+            ValidIssuer = _issuer,
+            ValidAudience = _audience,
+            IssuerSigningKey = _key
+        };
+
+        try
+        {
+            tokenHandler.ValidateToken(token, validationParameters, out _);
```

Comment:
it's a minor quibble, but I dont like that we are calling a library function "ValidateToken" inside a function called "ValidateToken"

Reply:
fixed